### DISCUSSION TITLE
Using tar_map() with tar_render()

### DISCUSSION THREAD

**Discussion by syntgx** [ropensci/tarchetypes#57] [Help]
Hello! 

Would you be so kind to show how tar_map() with tar_render() can be used? 
  1. How to generate multiple reports from 1 markdown file?
  2. How to generate multiple reports from multiple markdown files?

As a reference, I use the example from https://github.com/wlandau/targets-minimal

Thanks a lot! 

```
library(targets)
library(tarchetypes)
source("R/functions.R")
options(tidyverse.quiet = TRUE)


tar_option_set(packages = c("biglm", "dplyr", "ggplot2", "readr", "tidyr"))

values <- tibble::tibble(
  file_path = c("data/raw_data.csv", "data/raw_data_1.csv"),
  file_name = c("index.Rmd", "index_1.Rmd")
)


targets <- tar_map(
  values = values,
  tar_target(
    raw_data_file,
    file_path,
    format = "file"
  ),
  tar_target(
    raw_data,
    read_csv(raw_data_file, col_types = cols())
  ),
  tar_target(
    data,
    raw_data %>%
      filter(!is.na(Ozone))
  ),
  tar_target(hist, create_plot(data)),
  tar_target(fit, biglm(Ozone ~ Wind + Temp, data)),
  tar_render(report, "index.Rmd")
)


list(targets)

```

**Comment by wlandau** [comment]
This is an instance of https://github.com/ropensci/tarchetypes/issues/18. For multiple reports, I recommend writing something like this in your target list:

```{r}
tar_eval(
  values = list(
    name = rlang::syms(c("name1", "name2")),
    report_path = c("report1.Rmd", "report2.Rmd")
  ),
  tar_render(name, report_path)
)
```

For multiple output files from a single R Markdown report, you might have a look at parameterized R Markdown: https://books.ropensci.org/targets/files.html#parameterized-r-markdown

**Reply by syntgx** [reply]
For some peculiar reason, only the data from the first file_path ("data/raw_data.csv") is used. 
Another dataset is not presented in any of them. 

Thank you! 

```
library(targets)
library(tarchetypes)
source("R/functions.R")
options(tidyverse.quiet = TRUE)


tar_option_set(packages = c("biglm", "dplyr", "ggplot2", "readr", "tidyr"))

values <- tibble::tibble(
  file_path = c("data/raw_data.csv", "data/raw_data_1.csv"),
)


targets <- tar_map(
  values = values,
  tar_target(
    raw_data_file,
    file_path,
    format = "file"
  ),
  tar_target(
    raw_data,
    read_csv(raw_data_file, col_types = cols())
  ),
  tar_target(
    data,
    raw_data %>%
      filter(!is.na(Ozone))
  ),
  tar_target(hist, create_plot(data)),
  tar_target(fit, biglm(Ozone ~ Wind + Temp, data))
)


list(targets, 
     tar_eval(
       values = list(
         name = rlang::syms(c("name1", "name2")),
         report_path = c("index.Rmd", "index_1.Rmd")
       ),
       tar_render(name, report_path)
     )
  )
```

**Reply by wlandau** [reply]
I am not sure exactly what you mean. That `tar_map()` call above is correct, and it produces a collection of targets for each CSV input file. In order to be properly connected in the graph, the reports must call `tar_read()` and/or `tar_load()` on the appropriate targets after `tar_map()`. As explained at https://books.ropensci.org/targets/files.html#literate-programming, target dependencies of R Markdown reports come from the code chunks, and `tar_map()` does not modify those.
